internal/pkg/agent/install/install.go

```diff
@@ -46,26 +46,6 @@ func Install(cfgFile, topPath string, unprivileged bool, log *logp.Logger, pt *p
 		return utils.FileOwner{}, errors.New(err, "failed to discover the source directory for installation", errors.TypeFilesystem)
 	}
 
-	// We only uninstall Agent if it is currently installed.
-	status, _ := Status(topPath)
-	if status == Installed {
-		// Uninstall current installation
-		//
-		// There is no uninstall token for "install" command.
-		// Uninstall will fail on protected agent.
-		// The protected Agent will need to be uninstalled first before it can be installed.
-		pt.Describe("Uninstalling current Elastic Agent")
-		err = Uninstall(cfgFile, topPath, "", log, pt)
```

Comment:
Previously this call to uninstall was failing because `paths.Components()` would use a value that reflected the location from where `elastic-agent install -f` was being ran and not the installed agent.
This would result in the uninstall detecting 0 components, and skipping calling uninstall on each of the services, so endpoint uninstall was never called